server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/object/ObjectDeserializeOperator.java

```diff
@@ -0,0 +1,215 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.mpp.execution.operator.object;
+
+import org.apache.iotdb.db.mpp.execution.object.MPPObjectPool;
+import org.apache.iotdb.db.mpp.execution.object.ObjectEntry;
+import org.apache.iotdb.db.mpp.execution.object.ObjectEntryFactory;
+import org.apache.iotdb.db.mpp.execution.object.ObjectType;
+import org.apache.iotdb.db.mpp.execution.operator.Operator;
+import org.apache.iotdb.db.mpp.execution.operator.OperatorContext;
+import org.apache.iotdb.db.mpp.execution.operator.process.ProcessOperator;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.read.common.block.TsBlock;
+import org.apache.iotdb.tsfile.read.common.block.TsBlockBuilder;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static org.apache.iotdb.db.mpp.execution.operator.object.ObjectQueryConstant.BATCH_END_SYMBOL;
+import static org.apache.iotdb.db.mpp.execution.operator.object.ObjectQueryConstant.OBJECT_START_SYMBOL;
+import static org.apache.iotdb.tsfile.read.common.block.TsBlockBuilderStatus.DEFAULT_MAX_TSBLOCK_SIZE_IN_BYTES;
+
+public class ObjectDeserializeOperator implements ProcessOperator {
+
+  private final OperatorContext operatorContext;
+
+  private final String queryId;
+  private final MPPObjectPool objectPool = MPPObjectPool.getInstance();
+
+  private final List<TSDataType> outputDataTypes = Collections.singletonList(TSDataType.INT32);
+
+  private final Operator child;
+
+  private final List<ByteBuffer> bufferList = new ArrayList<>();
+
+  public ObjectDeserializeOperator(
+      OperatorContext operatorContext, String queryId, Operator child) {
+    this.operatorContext = operatorContext;
+    this.queryId = queryId;
+    this.child = child;
+  }
+
+  @Override
+  public OperatorContext getOperatorContext() {
+    return operatorContext;
+  }
+
+  @Override
+  public ListenableFuture<?> isBlocked() {
+    return child.isBlocked();
+  }
+
+  @Override
+  public TsBlock next() {
+    if (!hasNext()) {
+      throw new NoSuchElementException();
+    }
+
+    TsBlock tsBlock = child.next();
+    if (tsBlock == null || tsBlock.isEmpty()) {
+      return null;
+    }
+    ByteBuffer buffer;
+    for (int i = 0; i < tsBlock.getPositionCount() - 1; i++) {
+      buffer = ByteBuffer.wrap(tsBlock.getColumn(0).getBinary(i).getValues());
+      bufferList.add(buffer);
```

Comment:
Maybe it is not a safe way that sharing the intevmidie value by class level variable.